autotest/conftest.py

```diff
@@ -152,3 +152,17 @@ def pytest_configure(config):
             f"Attempting to run tests for GDAL {test_version} but library version is "
             f"{lib_version}. Do you need to run setdevenv.sh ?"
         )
+
+    import gdaltest
+
+    if not gdaltest.download_test_data():
```

Comment:
As the datasets are cached, if the user has already run the suite with GDAL_DOWNLOAD_TEST_DATA=YES, it is no longer necessary afterwards. Perhaps the pytest.skip() reason is enough ? Especially since this message will be at the beginning of the log, and the user is unlikely to see it unless he redirects it to a file and inspects the beginning.

Reply:
I see the message, because `pytest` takes about 20 seconds to collect tests. Then I can Ctrl-C and fix my environment. The `pytest.skip()` reason is only printed after all tests have run, and only then if the user called `pytest -r s`.

Maybe change the message to "tests relying on downloaded data _may_ be skipped" ?

That said, if it strikes you as visual clutter, let's take it out. I think the skip reason is the more useful component of this.